backend/src/routes/songs.js

```diff
@@ -1,21 +1,40 @@
 const express = require('express');
 const geofire = require('geofire-common');
+const moment = require('moment');
 const { firebase } = require('../firebase-init.js');
 
 const db = firebase.firestore();
 
 async function handleSong(req, res) {
   const { lat, lng, songId } = req.body;
+  const currTime = moment().unix();
   if (!lat || !lng || !songId) {
     return res.status(400).send('Insufficient info');
   }
+
   // TODO: Add data validation
   const result = await db.collection('songEntries').add({
     songId,
     locationHash: geofire.geohashForLocation([lat, lng]),
     lat,
     lng,
+    ts: currTime,
   });
+
+  const userRef = db.collection('user_metadata').doc(req.user_email);
+  try {
+    await db.runTransaction(async (t) => {
+      const doc = await t.get(userRef);
+      if (!doc.exists) {
+        return res.status(404).send('User not found');
+      }
+      const newSongs = doc.data().song_entries;
+      newSongs.push(result.id);
+      return t.update(userRef, { song_entries: newSongs });
```

Comment:
I did that originally and got some error regarding sending in a custom object to firebase; tried to debug but it still didn't work – it might have something to do with the geofire hash being stored. Either way, I just used the old way of using a transaction.

Reply:
Nevermind it works now I'm not seeing the error from yesterday